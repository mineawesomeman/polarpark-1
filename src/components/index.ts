export {default as Box} from '@app/components/general/Box/Box';
export {default as Button} from '@app/components/general/Button/Button';
export {default as ContentWrapper} from '@app/components/general/ContentWrapper/ContentWrapper';
export {default as Input} from '@app/components/general/Input/Input';
export {default as Icon} from '@app/components/general/Icon/Icon';
export {default as IconButton} from '@app/components/general/IconButton/IconButton';
export {default as Link} from '@app/components/general/Link/Link';
export {default as Placeholder} from '@app/components/general/Placeholder/Placeholder';
export {default as Section} from '@app/components/general/Section/Section';
export {default as Text} from '@app/components/general/Text/Text';

export {default as ComingSoon} from '@app/components/ui/ComingSoon/ComingSoon';
export {default as Event} from '@app/components/ui/Event/Event';
export {default as Footer} from '@app/components/ui/Footer/Footer';
export {default as Header} from '@app/components/ui/Header/Header';
export {default as ScheduleItem} from '@app/components/ui/ScheduleItem/ScheduleItem';
export {default as SignOnOptions} from '@app/components/ui/SignOnOptions/SignOnOptions';
export {default as TextDivider} from '@app/components/ui/TextDivider/TextDivider';
export {default as Ticket} from '@app/components/ui/Ticket/Ticket';
export {default as TicketPreview} from '@app/components/ui/TicketPreview/TicketPreview';
