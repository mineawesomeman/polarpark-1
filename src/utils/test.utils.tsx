import {ComponentType} from 'react';
import {ThemeProvider} from '@shopify/restyle';
import {render} from '@testing-library/react-native';
import theme from '@app/config/theme';
import '@testing-library/jest-native/extend-expect';

export interface WrapperProps {
  children: JSX.Element;
}

const Wrapper = ({children}: WrapperProps): JSX.Element => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

/* Override render from @testing-library/react-native */
const customRender = (ui: JSX.Element, options: any) =>
  render(ui, {wrapper: Wrapper as ComponentType, ...options});

/* Export the entire testing library */
export * from '@testing-library/react-native';

/* Export the custom render method */
export {customRender as render};
