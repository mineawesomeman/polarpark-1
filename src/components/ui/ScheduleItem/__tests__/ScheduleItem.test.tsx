import React from 'react';
import {render} from '@app/utils/test.utils';
import {ScheduleItem} from '@app/components';

describe('[ScheduleItem]', () => {
  it('should render correctly', () => {
    const component = render(
      <ScheduleItem
        description="This is a description"
        isoString="2023-03-31T16:05:00-04:00"
        title="This is a title"
      />,
      {},
    );
    expect(component).toMatchSnapshot();
  });
});
