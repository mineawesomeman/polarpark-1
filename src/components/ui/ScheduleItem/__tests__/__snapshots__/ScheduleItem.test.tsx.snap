// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ScheduleItem] should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "NunitoSans-Regular",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#14458E",
              },
            ]
          }
        >
          FRI MAR 31
        </Text>
         
        @ 
        4:05 PM
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "NunitoSans-Regular",
              "fontSize": 16,
              "fontWeight": "bold",
              "lineHeight": 24,
            },
          ]
        }
      >
        This is a title
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#A8A8A8",
              "fontFamily": "NunitoSans-Regular",
              "fontSize": 14,
              "fontWeight": "500",
              "lineHeight": 21,
            },
          ]
        }
      >
        This is a description
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 16,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 48,
              "width": 48,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
